core/synchronizer.go

```diff
@@ -616,6 +661,37 @@ func (s *Synchronizer) sendPing(ctx context.Context) {
 	}
 }
 
+func (s *Synchronizer) sendBatch(ctx context.Context, kind domain.Kind, batchType domain.BatchType, items domain.BatchItems) error {
+	event := domain.EventBatch
+	depth := ctx.Value(domain.ContextKeyDepth).(int)
+	msgId := ctx.Value(domain.ContextKeyMsgId).(string)
+	msg := domain.Batch{
+		Depth:     depth + 1,
+		Event:     &event,
+		Kind:      &kind,
+		MsgId:     msgId,
+		BatchType: string(batchType),
+		Items:     &items,
+	}
+	data, err := json.Marshal(msg)
+	if err != nil {
+		return fmt.Errorf("marshal batch message: %w", err)
+	}
+	err = s.outPool.Invoke(data)
+	if err != nil {
+		return fmt.Errorf("invoke outPool on batch message: %w", err)
+	}
+	clientId := utils.ClientIdentifierFromContext(ctx)
+	logger.L().Debug("sent batch message",
```

Comment:
TODO at some point we should see if it's possible to define default fields for logger in a given scope, I am tired of seeing/maintaining these
also one day deprecate zapper for slog

Reply:
Agree